### PR TITLE
Feature/django query set support

### PR DESCRIPTION
**Pull request by timtan**
# Purpose

* start to apply queryfilter to Django queryset
* just write two filter. make sure I am in correct way to add django support

# Changes

- a test project contains model only for test
- necessary settings for using django in test project
- some textfilter now can support django queryset

### CODE REVIEW

File: queryfilter/base.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(
 
     # TODO: Put it back when we need it
     # @abc.abstractmethod
```
Comment (davidleeplaysmart) [inline queryfilter/base.py]: we could remove the comment then!

File: queryfilter/base.py [hunk 2]
```diff
@@ -46,6 +46,27 @@ def false_with_drop_none_else_raise(self, field_name):
             return False
         raise FilterOnNoneValue(field_name)
 
+    def get_query_value(self):
+        return self.filter_args["value"]
```
Comment (davidleeplaysmart) [inline queryfilter/base.py]: hmmm ... unfortunately as I remember it's not always value 😅 It'll change according to type 🤔

Comment (timtan) [reply queryfilter/base.py]: ouch got it

File: queryfilter/base.py (hunk 1 above)
Comment (timtan) [reply queryfilter/base.py]: I hope we can remove the comment soon

File: queryfilter/base.py [hunk 3]
```diff
@@ -46,6 +46,27 @@ def false_with_drop_none_else_raise(self, field_name):
...
+
+
+class DjangoQueryFilterMixin(object):
+    __metaclass__ = abc.ABCMeta
+
+    def on_django_query(self, queryset):
+        from django.core.exceptions import FieldError
```
Comment (davidleeplaysmart) [inline queryfilter/base.py]: Is it good to import here? 🤔

Comment (timtan) [reply queryfilter/base.py]: if we import locally.   people who are not using django  will not be forced to install django.

File: queryfilter/tests/test_queryfilter.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
 from __future__ import absolute_import
 
 import pytest
+from django.test import TestCase
 
+from test_app.models import Data, DataWithoutName
```
[the remark this answers is not in the document]
Comment (davidleeplaysmart) [reply queryfilter/tests/test_queryfilter.py]: BTW, I thought DataWithoutNameField might be even better

**Pull request merged**